localstack/services/cloudformation/engine/transformers.py

```diff
@@ -50,27 +64,215 @@ def apply_transform_intrinsic_functions(
     stack_parameters: dict,
 ) -> dict:
     """Resolve constructs using the 'Fn::Transform' intrinsic function."""
-    from localstack.services.cloudformation.engine.template_deployer import resolve_refs_recursively
 
-    def _visit(obj, **_):
-        if isinstance(obj, dict) and obj.keys() == {"Fn::Transform"}:
-            transform = obj["Fn::Transform"]
+    def _visit(obj, path, **_):
+        if isinstance(obj, dict) and "Fn::Transform" in obj.keys():
+            transform = (
+                obj["Fn::Transform"]
+                if isinstance(obj["Fn::Transform"], dict)
+                else {"Name": obj["Fn::Transform"]}
+            )
             transform_name = transform.get("Name")
             transformer_class = transformers.get(transform_name)
+            macro_store = get_cloudformation_store(account_id, region_name).macros
+            parameters = transform.get("Parameters") or {}
+            parameters = resolve_refs_recursively(
+                account_id,
+                region_name,
+                stack_name,
+                resources,
+                mappings,
+                conditions,
+                stack_parameters,
+                parameters,
+            )
             if transformer_class:
                 transformer = transformer_class()
-                parameters = transform.get("Parameters") or {}
-                parameters = resolve_refs_recursively(
-                    account_id,
-                    region_name,
-                    stack_name,
-                    resources,
-                    mappings,
-                    conditions,
-                    stack_parameters,
-                    parameters,
-                )
                 return transformer.transform(account_id, region_name, parameters)
+
+            elif transform_name in macro_store:
+                obj_copy = dict(obj)
+                obj_copy.pop("Fn::Transform")
+                result = execute_macro(
+                    account_id, region_name, obj_copy, transform, stack_parameters, parameters, True
+                )
+                return result
+            else:
+                LOG.warning("Unsupported transform function: %s", transform_name)
```

Comment:
nit: I'd suggest adding the stack name to the log statement for easier debugging